src/prediction/predict_race.py

```diff
@@ -0,0 +1,35 @@
+from dataset.load_dataset import TrainDataModule
+from PIL import Image, UnidentifiedImageError
+import requests
+import numpy as np
+from deepface import DeepFace
+from typing import Any, Dict
+
+class RacePrediction():
+    def __init__(self):
+        self.train_dataset = TrainDataModule().train_dataset
```

Comment:
Let's pass the dataset as a constructor argument and decouple the predictor and the dataset. This class should not be responsible with creating the dataset/datamodule object.

Do you think you will also need other datasets (e.g. validation/test datasets) from the same datamodule? If you only need the train dataset, maybe it's worth constructing just that and passing it to the constructor.
```train_dataset=ImageDataset(raw_dataset['train']['SAMPLE_ID'], raw_dataset['train']['TEXT'], raw_dataset['train']['URL'])
race_predictor = RacePredictor(dataset=train_dataset)```